### PR TITLE
JAMES-3498 Avoid mailbox in EventBus RabbitMQ resources

### PR DESCRIPTION
**Pull request by chibenwa**
- [x] Waiting a green build before being opened for review
 - [x] Write a test, starting 2 RabbitMQ eventBus with different naming strategy and ensures they are well isolated.

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

(unrelated)

File: event-bus/distributed/src/main/java/org/apache/james/events/NamingStrategy.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package org.apache.james.events;
```
Comment (Arsnael) [inline event-bus/distributed/src/main/java/org/apache/james/events/NamingStrategy.java:1]: Missing license

File: event-bus/distributed/src/main/java/org/apache/james/events/NamingStrategy.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+public class NamingStrategy {
+    private final String prefix;
+
+    public NamingStrategy(String prefix) {
+        this.prefix = prefix;
+    }
+
+    public RegistrationQueueName queueName(EventBusId eventBusId) {
+        return  new RegistrationQueueName(prefix + "-eventbus-" + eventBusId.asString());
```
Comment (Arsnael) [inline event-bus/distributed/src/main/java/org/apache/james/events/NamingStrategy.java:13]: extra space

**Review by Arsnael — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**